### PR TITLE
Return existing CAT wallet instead of raising

### PR DESCRIPTION
**Pull request by Quexington**
There's no reason to raise here, all the caller is looking for is an instance of the CATWallet that they're looking for and to create it if it doesn't exist.

https://github.com/Chia-Network/chia-blockchain/issues/14002

### CODE REVIEW

**Comment by arvidn** [comment]
I think it would be good to rename the function to indicate that it has this new semantic. Perhaps `get_or_create_cat_wallet()` or something like that.
I think that would also requiring updating all call sites, which helps a reviewer to also ensure these new semantics make sense in all cases where it's currently called.

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ async def create_wallet_for_cat(
                 assert isinstance(w, CATWallet)
                 if w.get_asset_id() == limitations_program_hash_hex:
                     self.log.warning("Not creating wallet for already existing CAT wallet")
-                    raise ValueError("Wallet already exists")
+                    return w
```
Comment (Quexington) [inline chia/wallet/cat_wallet/cat_wallet.py:195]: Here's the main change

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -586,7 +586,7 @@ async def create_new_wallet(self, request: Dict) -> EndpointResult:
 
             elif request["mode"] == "existing":
                 async with self.service.wallet_state_manager.lock:
-                    cat_wallet = await CATWallet.create_wallet_for_cat(
+                    cat_wallet = await CATWallet.get_or_create_wallet_for_cat(
                         wallet_state_manager, main_wallet, request["asset_id"], name
                     )
                 self.service.wallet_state_manager.state_changed("wallet_created")
```
Comment (emlowe) [inline chia/rpc/wallet_rpc_api.py:592]: this state changed message will fire now in cases where the wallet wasn't actually created, but perhaps that is harmless

Comment (Quexington) [reply chia/rpc/wallet_rpc_api.py]: Moved the notification inside of the function just in case

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 3]
```diff
@@ -208,6 +208,7 @@ async def create_wallet_for_cat(
 
         self.lineage_store = await CATLineageStore.create(self.wallet_state_manager.db_wrapper, self.get_asset_id())
         await self.wallet_state_manager.add_new_wallet(self, self.id())
+        self.wallet_state_manager.state_changed("wallet_created")
```
Comment (emlowe) [inline chia/wallet/cat_wallet/cat_wallet.py:211]: Interestingly, `add_new_wallet` already calls `state_changed` - not sure if all the previous calls were also redundant. https://github.com/Chia-Network/chia-blockchain/blob/c209f886b1e8c30902c0b6e05884d55ad1c9591f/chia/wallet/wallet_state_manager.py#L1687

**Review by emlowe — APPROVED** [review]
Looks good to me

File: chia/wallet/cat_wallet/cat_wallet.py (hunk 1 above)
Comment (wallentx) [reply chia/wallet/cat_wallet/cat_wallet.py:195]: Resolving, as this has been reviewed, and approved

**Pull request merged**